cpukit/score/src/threadsetpriority.c

```diff
@@ -30,13 +30,15 @@ static bool _Thread_Set_priority_filter(
   Priority_Control  new_priority;
   Priority_Control *old_priority_ptr;
 
-  current_priority = the_thread->current_priority;
+  current_priority = the_thread->Priority_node.current_priority;
   new_priority = *new_priority_ptr;
 
-  old_priority_ptr = arg;
-  *old_priority_ptr = current_priority;
+  if (arg != NULL) {
```

Comment:
Is this NULL check required?

Reply:
Nope. You are correct. The check is not required.